File: fizzbuzz.hs
module Main where

fb :: Integer -> String
fb n | n `mod` 15 == 0 = "FizzBuzz"
     | n `mod`  5 == 0 = "Buzz"
     | n `mod`  3 == 0 = "Fizz"
     |       otherwise = show n

main = print $ map fb [1..1000000]

File: fizzbuzz.c
#include <stdio.h>

int
main()
{
	int i;

	for (i = 0; i <= 1000000; i++) {
		if (!(i % 15))
			puts("FizzBuzz");
		else if (!(i % 5))
			puts("Buzz");
		else if (!(i % 3))
			puts("Fizz");
		else
			printf("%d\n", i);
	}

	return 0;
}

